design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,257 @@
+# Spike - Containerless CLI
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**                                              | **Description**                                              |
+| ----------------------------------------------------- | ------------------------------------------------------------ |
+| Epic                                                  | https://github.com/cyberark/conjur/issues/1892               |
+| Ruby CLI                                              | https://github.com/cyberark/conjur-cli                       |
+| Go CLI                                                | https://github.com/cyberark/conjur-cli-go                    |
+| Python CLI                                            | https://github.com/cyberark/conjur-api-python3               |
+| [FPM](https://github.com/jordansissel/fpm)            | For building packages for multiple platforms (RPM, DEB, OS X) |
+| [Debify](https://github.com/conjurinc/debify)         | Build a package from a Ruby gem                              |
+| [Ruby packager](https://github.com/pmq20/ruby-packer) | Packs Ruby application into a single executable for increased portability |
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+This document is one piece in the bigger piece of the Dockerless puzzle. In this document we will discuss possible options for a containerless CLI and present a recommendation for which one we should pursue.
+
+In addition, regardless of how Conjur is deployed delivering and officially supporting a CLI that does not require Docker allows our solution to be portable and more easily attainable.
+
+#### Requirements
```

Comment:
Here is the most important requirement: we only have a single CLI for customers.  So if something other than Ruby is desired, we need to have a discussion about moving to that new one and away for Ruby.

Reply:
Isn't that how it is now? or are you saying deprecate all other CLIs except for the one we chose?

Reply:
@sigalsax: Yes, exactly. We only have a single GA, officially support CLI.  If this project selects a different one to meet its requirements, it should replace Ruby for all customers.  For this reason, the consequences of picking Python or Go should be clear.  It's important that @InbalZilberman and Sharon see this requirement and let me know if they disagree.

Reply:
I remember this point being mentioned that the ultimate goal is to replace the Ruby CLI (if that is the direction we want to go in) or further developer it. Either way there should only be one that is distributed. @InbalZilberman ?